runtime/lua/vim/lsp/buf.lua

```diff
@@ -202,6 +202,65 @@ function M.outgoing_calls()
   end)
 end
 
+--- List workspace folders.
+function M.list_workspace_folders()
+  for _, client in ipairs(vim.lsp.buf_get_clients()) do
+    for _, folder in ipairs(client.workspaceFolders) do
+      print(folder.name)
+    end
+  end
+end
+
+--- Add a workspace folder.
+function M.add_workspace_folder(workspace_folder)
+  workspace_folder = workspace_folder or npcall(vfn.input, "Workspace Folder: ", vfn.expand('%:p:h'))
+  api.nvim_command("redraw")
+  if not (workspace_folder and #workspace_folder > 0) then return end
+  if vim.fn.isdirectory(workspace_folder) == 0 then
+    print(workspace_folder, " is not a valid directory")
+    return
+  end
+  local params = util.make_workspace_params()
+  local workspace_folder_entry = {uri = vim.uri_from_fname(workspace_folder); name = workspace_folder}
+  table.insert(params.event.added, workspace_folder_entry )
+  for _, client in ipairs(vim.lsp.buf_get_clients()) do
+    for idx, folder in ipairs(client.workspaceFolders) do
+      local found = false
+      if folder.name == workspace_folder then
+        found = true
```

Comment:
Should probably return when found, since you don't want to print this a million times, right?

Reply:
I added a break statement since it still needs to loop over every client